spec/rspec/core/bisect/runner_spec.rb

```diff
@@ -213,19 +213,67 @@ def repro_command_from(ids)
       open3_method = Open3.respond_to?(:capture2e) ? :capture2e : :popen3
       open3_method = :popen3 if RSpec::Support::Ruby.jruby?
 
+      def called_environment
+        @called_environment
+      end
+
+      if open3_method == :capture2e
+        RSpec::Matchers.define :have_invoked_command_with_env do |command, environment|
+          match do
+            expect(Open3).to have_received(open3_method).with(environment, command)
+          end
+        end
+      elsif open3_method == :popen3
+        RSpec::Matchers.define :have_invoked_command_with_env do |command, environment|
+          match do
+            expect(Open3).to have_received(open3_method).with(command)
+            expect(called_environment).to include(environment)
+          end
+        end
+      end
+
       before do
-        allow(Open3).to receive(open3_method).and_return(
+        allow(Open3).to receive(open3_method) do
+          @called_environment = ENV.to_hash.dup
           [double("Exit Status"), double("Stdout/err")]
-        )
+        end
+
         allow(server).to receive(:capture_run_results) do |&block|
           block.call
           "the results"
         end
       end
 
       it "runs the suite with the original CLI options" do
-        runner.original_results
-        expect(Open3).to have_received(open3_method).with(a_string_including("--seed 1234"))
+        expect(
+          runner.original_results
+        ).to have_invoked_command_with_env(a_string_including("--seed 1234"), {})
+      end
+
+      context 'when --bisect is present in SPEC_OPTS' do
+        it "runs the suite with --bisect removed from the environment" do
+          expect(
+            with_env_vars('SPEC_OPTS' => '--bisect --fail-fast') do
+              runner.original_results
+            end
+          ).to have_invoked_command_with_env(
+            a_string_including("--seed 1234"),
+            { 'SPEC_OPTS' => '--fail-fast' }
+          )
+        end
+      end
+
+      context 'when --bisect=verbose is present in SPEC_OPTS' do
+        it "runs the suite with --bisect removed from the environment" do
+          expect(
+            with_env_vars('SPEC_OPTS' => '--bisect=verbose --fail-fast') do
+              runner.original_results
+            end
+          ).to have_invoked_command_with_env(
+            a_string_including("--seed 1234"),
+            { 'SPEC_OPTS' => '--fail-fast' }
+          )
+        end
```

Comment:
I see what you mean about it being tricky to figure out what to treat as the subject for these expectations.  Maybe the block approach does make the most sense.  I'm happy to merge with this has it is but if you want to switch back to the block form that would be fine, too.